grantnav/frontend/views.py

```diff
@@ -491,6 +491,185 @@ def add_advanced_search_information_in_context(context):
     return context
 
 
+def term_facet_from_parameters(request, json_query, field_name, param_name, bool_index, field, is_json=False):
+    new_filter = []
+    for value in request.GET.getlist(param_name):
+        new_filter.append({"term": {field_name: value}})
+
+    json_query["query"]["bool"]["filter"][bool_index]["bool"]["should"] = new_filter
+
+
+def amount_facet_from_parameters(request, json_query):
+    new_filter = []
+    for value in request.GET.getlist('amountAwarded'):
+        for item in FIXED_AMOUNT_RANGES:
+            if value == str(item['from']):
+                new_range = {"gte": item["from"]}
+                to_ = item.get("to")
+                if to_:
+                    new_range["lt"] = to_
+                new_filter.append({"range": {"amountAwarded": new_range}})
+
+    json_query["query"]["bool"]["filter"][2]["bool"]["should"] = new_filter
+
+
+def date_facet_from_parameters(request, json_query):
+    new_filter = []
+    for value in request.GET.getlist("awardDate"):
+        new_filter.append(
+            {"range": {
+                "awardDate": {"format": "year",
+                                  "gte": value + "||/y",
+                                  "lte": value + "||/y"}
+            }
+            }
+        )
+
+    json_query["query"]["bool"]["filter"][4]["bool"]["should"] = new_filter
+
+
+def term_facet_size_from_parameters(request, json_query):
+    try:
+        aggs = json_query["aggs"]
+    except KeyError:
+        aggs = BASIC_QUERY['aggs']
+
+    for agg_name, agg in aggs.items():
+        if "terms" not in agg:
+            continue
+        more = request.GET.get(agg_name+'More')
+        if more:
+            agg["terms"]["size"] = 50
+        else:
+            agg["terms"]["size"] = 3
+
+def non_term_facet_size_from_parameters(request, json_query, agg_name):
+
+    more = request.GET.get(agg_name+'More')
+    if more:
+        json_query['extra_context'][agg_name + '_facet_size'] = 50
+    else:
+        json_query['extra_context'][agg_name + '_facet_size'] = 3
+
+
+def create_json_query_from_parameters(request):
```

Comment:
Would be nice to have a doc string for this function as it is now heavily used. `parameters` and `json_query` are quite generic terms so for someone not familiar with the code it'd be nice to explain `json_query` =  the elastic search json query and `parameters`  = GET parameters from the request.